R/declare_potential_outcomes.R

```diff
@@ -290,54 +290,32 @@ pofdv <- function(design, i, step){
   }
   
   unassigned_vars <- check("assignment_variables", "assignment", "assignment_variables", from = i + 1)
-  unrevealed_vars <- check("assignment_variables", "reveal_outcomes", "outcome_variables", from = i + 1)
+  unrevealed_vars <- check("assignment_variables", "reveal", "outcome_variables", from = i + 1)
   
   cant_find <- prev_unassigned %i% prev_unrevealed %i% unassigned_vars %i% unrevealed_vars
   
-  if (length(cant_find) > 0) {
-    # warning(
-    #   "Assignment variables (", paste(cant_find, sep = ", "),
-    #   ") were declared in a potential outcomes step (", attr(step, "label"),
-    #   "), but never later assigned by an assignment step.", call. = FALSE)
-  } else {
-    
-    new_step <- eval_tidy(quo(declare_reveal(outcome_variables = !!this_step_meta$outcome_variables,
-                                             assignment_variables = !!this_step_meta$assignment_variables,
-                                             label = !!paste("Autogenerated by", attr(step, "label"))) ))
-    attr(new_step, "auto-generated") <- TRUE
-    
-    for (step_j in design[length(design):(i + 1)]) {
-      if (is.null(step_meta <- attr(step_j, "step_meta"))) next;
-      if (attr(step_j, "step_type") == "assignment") {
-        if (any(step_meta$assignment_variables %in% attr(step, "step_meta")$assignment_variables)) {
-          # warning("Attempting to inject a `declare_reveal(", this_step_meta$outcome_variables, ", ",
```

Comment:
The warnings should probably come back at some point.

Reply:
thx, yes, will remember to come back to this